app/src/main/java/com/aguesoftguar/medalarm/addeditmedicine/AddEditMedicineFragment.java

```diff
@@ -593,18 +602,63 @@ public void onClick(View view) {
       inflater.inflate(R.menu.add_and_edit_medicine_settings, menu);
    }
 
+   @Override
+   public boolean onOptionsItemSelected(MenuItem item) {
+      switch (item.getItemId()) {
+
+         case R.id.save_medicine:
+            LayoutInflater inflater =
+               LayoutInflater.from(AddEditMedicineFragment.this.getContext());
+            View contentView = inflater.inflate(R.layout.activity_add_and_edit_medicine, null);
+
+            EditText medicineNameEditText =
+                    (EditText) contentView.findViewById(R.id.medicine_name_edittext);
```

Comment:
This doesn't work because you has inflated the activity view into the fragment view again. You need to substitute the code for the following one:
`EditText medicineNameEditText = (EditText) getActivity().findViewById(R.id.medicine_name_edittext);`